tests/integration_tests/modules/test_power_state_change.py

```diff
@@ -27,13 +27,13 @@ def _detect_reboot(instance: IntegrationInstance):
     # detecting the first boot or second boot, so we also check
     # the logs to ensure we've booted twice. If the logs show we've
     # only booted once, wait until we've booted twice
```

Comment:
Total aside, but I just noticed that this test could fail (by exiting early) if run on an instance which has booted more than once without cloud-init.log removal: the "first" boot would have multiple entries. Generally for this sort of check, I think we need a "before" count which is expected to increase before we consider the reboot complete (and this is why we should capture this in library code, so every .restart() call DTRT).

Anyway, moving on.